youtube_dl/extractor/telegram.py

```diff
@@ -0,0 +1,60 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+)
+
+
+class TelegramIE(InfoExtractor):
+    _VALID_URL = r'https://t\.me/(?P<user>[^/]+)/(?P<id>\d+)'
```

Comment:
I suppose any http URL gets redirected:
```suggestion
    _VALID_URL = r'https?://t\.me/(?P<user>[^/]+)/(?P<id>\d+)'
```